### PR TITLE
Restrict flash loan if vat is not live

### PR DESCRIPTION
**Pull request by gbalabasquer**

### CODE REVIEW

File: src/flash.sol [hunk 1]
```diff
@@ -117,14 +118,17 @@ contract DssFlash is IERC3156FlashLender, IVatDaiFlashLender {
             return 0;
         }
     }
+
     function flashFee(
         address token,
         uint256 amount
     ) external override view returns (uint256) {
+        amount;
```
Comment (hexonaut) [inline src/flash.sol:126]: You can just exclude the parameter name in the function definition.

**Review by hexonaut — APPROVED** [review]
LGTM. Slight preference to exclude the `amount` instead of sticking it in as a no-op.

File: src/flash.sol (hunk 1 above)
Comment (gbalabasquer) [reply src/flash.sol:126]: Yes, but that will remove it from the automatic generated abi. So considered it was better this way for this specific case.

**Review by kmbarry1 — APPROVED** [review]
LGTM

File: src/flash.t.sol [hunk 2]
```diff
@@ -415,93 +415,103 @@ contract DssFlashTest is DSTest {
         assertEq(vat.sin(address(flash)), 0);
     }
 
+    function testFail_flash_vat_not_live() public {
+        vat.cage();
```
Comment (brianmcmichael) [inline src/flash.t.sol:419]: This is definitely one of those places where foundry's expect fail checks will help, if the test reverts on these vat calls the tests still pass

**Review by brianmcmichael — APPROVED** [review]
:+1:

File: src/flash.t.sol (hunk 2 above)
Comment (gbalabasquer) [reply src/flash.t.sol:419]: Yep, we can still do with dapptools and those functions I created for `ds-test`.

**Pull request merged**